shared/src/formationData.ts

```diff
@@ -69,8 +70,14 @@ export interface FormationData {
   readonly formationResponse: FormationSubmitResponse | undefined;
   readonly getFilingResponse: GetFilingResponse | undefined;
   readonly completedFilingPayment: boolean;
+  readonly businessNameSearch: BusinessNameSearch | undefined;
 }
 
+export type BusinessNameSearch = {
+  businessNameAvailability: NameAvailability | undefined;
+  hasBusinessNameBeenSearched: boolean | undefined;
```

Comment:
how do we use this `hasBusinessNameBeenSearched` variable? 

1. Why does `hasBusinessNameBeenSearched` have type `boolean | undefined`? If it is not yet searched, I imagine it would be false. Why is undefined a possible value?
2. Although, do we even need it at all? Isn't it always the same as the `businessNameAvailability` value? ie, if `businessNameAvailability` is undefined, it has not been searched. If `businessNameAvailability` is not undefined, it has been searched